irods/test/meta_test.py

```diff
@@ -646,6 +647,22 @@ def test_thoroughgoing_correctness_of_genquery_result_column_mappings__issue_642
         finally:
             del test_coll.metadata[meta_key]
 
+    def test_xml_mode_addresses_odd_metadata_characters__issue_582(self):
+        session = self.sess
+        try:
+            # Character '\x01' is unprintable and messes up the default(STANDARD_XML) parser.
+            hexvalue = "53:6b:6f:70:65:43:61:6c:58:c3:af:c2:bf:c2:bd:c3:af:c2:bf:c2:bd:23:01:32:64:31"
+            hex_list = [int(byte, 16) for byte in hexvalue.split(":")]
+            string_value = "".join([chr(byte) for byte in hex_list])
+
+            from irods.helpers import xml_mode
+
+            with xml_mode('QUASI_XML'):
+                obj = session.collections.get('/tempZone/home/rods')
+                attr_str = "awesome"
```

Comment:
Consider defining this at the top of the test method so that the `finally` doesn't blow up should an `Exception` be `raise`d before this line (i.e. `attr_str` may not be defined when line 665 is executed).